keras/engine/topology.py

```diff
@@ -2831,7 +2834,7 @@ def preprocess_weights_for_loading(layer, weights,
                 weights[0] = np.transpose(weights[0], (2, 3, 0, 1))
 
         if layer.__class__.__name__ == 'Conv3D':
-            if layer.data_format == 'channels_first':
+            if layer.data_format == 'channels_first' and original_backend in {'th', 'theano'}:
```

Comment:
Not if the old `dim_ordering` was `'tf'`:
https://github.com/fchollet/keras/blob/1.2.2/keras/layers/convolutional.py#L422
Transposing in this case messes things up and is the main source of failure.